### PR TITLE
EFA-491 Fix recipient issues due to threading issues

### PR DESCRIPTION
**Pull request by ig-garcia**
It seems like there were some coroutine misusing issues like abusing of runBlocking usage.

### CODE REVIEW

**Comment by Stereodark** [comment]
wow this is a massive PR, I see you added tons of suspend statements to make the calls non-blocking (async), it will take some time and lots of testing to see if there is any race condition. It will be almost impossible just reading the code to find out. I will take my time to check all the changes, but I think testing thoroughly will be the only way to ensure there are no regressions.

**Comment by ig-garcia** [comment]
> wow this is a massive PR, I see you added tons of suspend statements to make the calls non-blocking (async), it will take some time and lots of testing to see if there is any race condition. It will be almost impossible just reading the code to find out. I will take my time to check all the changes, but I think testing thoroughly will be the only way to ensure there are no regressions.



> wow this is a massive PR, I see you added tons of suspend statements to make the calls non-blocking (async), it will take some time and lots of testing to see if there is any race condition. It will be almost impossible just reading the code to find out. I will take my time to check all the changes, but I think testing thoroughly will be the only way to ensure there are no regressions.

Yes, what happened is that initially there were many calls from Java to Kotlin Coroutines, then using runBlocking is the only way to connect this code. But using it within the coroutines themselves is not recommended and it seemed to be causing some issues. In addition, an instance in AutoCloseableThread seemed not to be being freed after thread run, which may have been causing a memory leak.

Regarding race conditions of course I don't expect having any, because we need to understand that runBlocking blocks the current thread. And I just replaced these calls with non-blocking coroutines respecting the execution order etc.
This said, errors are inherent to human so still of course there might be some issue.

File: k9mail/src/main/java/com/fsck/k9/planck/infrastructure/threading/AutoCloseableThread.kt [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
 open class AutoCloseableThread<T : AutoCloseable>(
-    private val closeable: T,
+    private var closeable: T?,
     target: Runnable?
 ) : Thread(target) {
     override fun run() {
         closeable.use {
             super.run()
         }
+        closeable = null
```
Comment (Stereodark) [inline k9mail/src/main/java/com/fsck/k9/planck/infrastructure/threading/AutoCloseableThread.kt:11]: is this needed? As far as I know the .use is already taking care of closing the closable, were we concerned about memory leaks? The one thing that might happen is unhandled exceptions, but for these the closable = null will not help.

**Review by Stereodark — APPROVED** [review]
Looks good, as I shared hard to foresee any potential issue just by reading the code for this one.

File: k9mail/src/main/java/com/fsck/k9/planck/infrastructure/threading/AutoCloseableThread.kt (hunk 1 above)
Comment (ig-garcia) [reply k9mail/src/main/java/com/fsck/k9/planck/infrastructure/threading/AutoCloseableThread.kt:11]: The thing is, the .use closes the closeable, but what I am doing here is freeing the reference to the closeable which is a different thing.

**Review by pzplanck — APPROVED** [review]

**Pull request merged**